src/Gmsh.jl

```diff
@@ -1,18 +1,44 @@
-module Gmsh
+# Copied from https://github.com/gridap/GridapGmsh.jl/blob/master/deps/build.jl
+# to use the new officially registered Gmsh JLL
 
-const deps_path = abspath(joinpath(dirname(@__FILE__), "..", "deps"))
+module Gmsh
 
-if !isfile(joinpath(deps_path, "deps.jl"))
-    error("Gmsh not installed properly, run Pkg.build(\"Gmsh\"), restart Julia and try again")
-end
-include(joinpath(deps_path, "deps.jl"))
-include(joinpath(deps_path, "usr", "lib", "gmsh.jl"))
-global const api = gmsh
+using Libdl
 
-function __init__()
-    # Always check your dependencies that live in `deps.jl`
-    check_deps()
+deps_jl = joinpath(@__DIR__, "..", "deps", "deps.jl")
+if !isfile(deps_jl)
+  s = """
+  Package GridapGmsh not installed properly.
+  Run Pkg.build(\"GridapGmsh\"), restart Julia and try again.
+  """
+  error(s)
 end
 
+include(deps_jl)
 
+if GMSH_FOUND
+    include(gmsh_jl)
+    # Hack taken from MPI.jl
+    function __init__()
+      @static if Sys.isunix()
+        Libdl.dlopen(gmsh.lib, Libdl.RTLD_LAZY | Libdl.RTLD_GLOBAL)
```

Comment:
Can you explain what's happening here? I have the feeling things could be simplified a little: https://github.com/Ferrite-FEM/FerriteGmsh.jl/blob/master/src/FerriteGmsh.jl#L5-L7

Reply:
Interesting. No, I can't explain.
But if your much shorter code works that's surely preferable.